plugins/categoryaslink/class.categoryaslink.plugin.php

```diff
@@ -0,0 +1,176 @@
+ */
+class CategoryAsLinkPlugin extends Gdn_Plugin {
+
+    /**
+     * Executes every time the plugin is turned on, makes changes to config.
+     */
+    public function setup() {
+        $this->structure();
+    }
+
+
+    /**
+     * Add columns to the Category table to stor Linked CategoryIDs or LinkedDiscussion URLs.
+     */
+    public function structure() {
+        Gdn::structure()->table('Category')
+            ->column('LinkedCategoryID', 'int', true)
+            ->column('LinkedDiscussion', 'varchar(255)', true)
+            ->set();
+    }
+
+
+    /**
+     * On the category add/edit page the dashboard, add a dropdown of categories, to link a category
+     * to another category. Add a textbox to link a category to some other URI (e.g. a discussion).
+     *
+     * @param SettingsController $sender
+     * @param SettingsController $args
+     */
+    public function settingsController_addEditCategory_handler($sender, $args) {
+        $sender->Data['_ExtendedFields']['LinkedCategoryID'] = [
+            'Control' => 'categorydropdown',
+            'Description' => 'Act as a copy of another category on this forum.',
+            'Options' => ['IncludeNull' => 'None']
+        ];
+        $sender->Data['_ExtendedFields']['LinkedDiscussion'] = [
+            'Control' => 'TextBox',
+            'Description' => 'Act as a link to a discussion. Paste the full URL of the address. If you have linked this category to another category this will override it.',
+            'Options' => ['IncludeNull' => 'None']
+        ];
+
+        $sender->Form->validateRule('LinkedDiscussion', 'validateWebAddress', t('Linked Discussion must be a valid Web Address'));
+    }
+
+
+    /**
+     * Loop through all the categories and prepare data to display LinkedCategory data or LinkedDiscussions.
+     *
+     * @param CategoriesController $sender
+     * @param CategoriesController $args
+     */
+    public function categoriesController_render_before($sender, $args) {
+        if (!isset($sender->Data['CategoryTree'])
+            || !is_array($sender->Data['CategoryTree'])
+            || !is_array(reset($sender->Data['CategoryTree']))) {
+            return;
+        }
+
+        $categories =& $sender->Data['CategoryTree'];
+
+        foreach ($categories as &$category) {
+            if (!val('LinkedCategoryID', $category) && !val('LinkedDiscussion', $category)) {
+                continue;
+            }
+
+            $linkedCategoryID = val('LinkedCategoryID', $category);
+            if (isset($linkedCategoryID)) {
+                // Get linked category data.
+                $linkedCategory[] = $sender->CategoryModel->categories($linkedCategoryID);
+                $sender->CategoryModel->joinRecent($linkedCategory, $linkedCategoryID);
+                $linkedCategory = $linkedCategory[0];
+                $category['_CssClass'] = 'Aliased AliasedCategory';
```

Comment:
Couldn't a lot of this be replaced with an assignedment or a clone?

```php
if (is_array($linkedCategory)) {
    $category = $linkedCategory;
} elseif (is_object($linkedCategory) {
    $category = clone $linkedCategory;
}
```

All of the actual values are strings or numbers so they will all get copied by value if you just assign an array, and they can cloned with the clone keyword.

Reply:
Thanks, fixed it by just cloning the array. It is never an object. Also, I realized that according to the use case, you create an alias to a category because you want it to have a different name and description.